packages/kit/src/vite/dev/index.js

```diff
@@ -180,13 +180,44 @@ export async function dev(vite, vite_config, svelte_config) {
 		extensions: []
 	});
 
+	vite.middlewares.use(async (req, res, next) => {
+		try {
+			if (!req.url || !req.method) throw new Error('Incomplete request');
```

Comment:
is this actually a thing that can happen? if i remove this line and get rid of the `try-catch`, TypeScript doesn't complain

Reply:
No, I don't think we need to worry about it. I had copied it from the main middleware. I've removed it in both places. The `try`/`catch` on the otherhand can't be removed

Reply:
Are you sure? I don't see any code there that's in danger of throwing

Reply:
If you try to remove it there's a test that fails with the error below showing where it can throw. It must bring down the whole server because all subsequent tests then fail.

```
[WebServer] file:///home/bmccann/src/kit/packages/kit/src/vite/dev/index.js:188
			const decoded = decodeURI(new URL(base + req.url).pathname);
			                ^

URIError: URI malformed
    at decodeURI (<anonymous>)
    at packages/kit/src/vite/dev/index.js:188:20
    at call (node_modules/.pnpm/vite@3.0.0/node_modules/vite/dist/node/chunks/dep-561c5231.js:45966:7)
    at next (node_modules/.pnpm/vite@3.0.0/node_modules/vite/dist/node/chunks/dep-561c5231.js:45910:5)
    at Function.handle (node_modules/.pnpm/vite@3.0.0/node_modules/vite/dist/node/chunks/dep-561c5231.js:45913:3)
    at Server.app (node_modules/.pnpm/vite@3.0.0/node_modules/vite/dist/node/chunks/dep-561c5231.js:45778:37)
    at Server.emit (node:events:527:28)
    at parserOnIncoming (node:_http_server:956:12)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:128:17)
```

Reply:
ah, gotcha. fair enough